LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,496 @@
+                string registryWithImageName = $"{imageDetails.Registry}/{imageDetails.ImageName}";
+                Progress<JSONMessage> theMess = new Progress<JSONMessage>(
+                    msg =>
+                    {    
+                        Console.WriteLine($"{msg.Status}|{msg.ProgressMessage}|{msg.ErrorMessage}");
+                    });
+
+                await client.Images.PushImageAsync(
+                    registryWithImageName,
+                    new ImagePushParameters
+                    {
+                        Tag = imageDetails.ImageTag,
+                    },
+                    new AuthConfig
+                    {
+                        Username = ContainerRegistryCredentialsRes.Result.Username,
+                        Password = ContainerRegistryCredentialsRes.Result.Password
+                    },
+                    theMess
+                );
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                Environment.Exit(1);
+            }
+        }
+
+        public async Task CheckAndUploadLinuxContainerImageAsync(string imageSkipToken = null, int pageSize = 10)
+        {
+            do
+            {
+                var ContainerImagesRequest = new ListContainerImagesRequest
+                {
+                    SkipToken = imageSkipToken,
+                    PageSize = pageSize
+                };
+                var ContainerImagesResponse = await PlayFabMultiplayerAPI.ListContainerImagesAsync(ContainerImagesRequest);
+
+                ErrorCheck(ContainerImagesResponse.Error);
+
+                imageSkipToken = ContainerImagesResponse.Result.SkipToken ?? null;
+                IEnumerable<string> existingImageNames = ContainerImagesResponse.Result.Images.Where(x => x == settings.ContainerStartParameters.ImageDetails.ImageName);
+
+                if (!existingImageNames.Any() && string.IsNullOrEmpty(imageSkipToken))
+                {
+                    await UploadLinuxContainerImageAsync(settings.ContainerStartParameters.ImageDetails);
+                }
+                else if (existingImageNames.Any())
+                {
+                    var ContainerImageTagsRequest = new ListContainerImageTagsRequest
+                    {
+                        ImageName = existingImageNames.First() ?? null,
+                    };
+                    var ContainerImageTagsResponse = await PlayFabMultiplayerAPI.ListContainerImageTagsAsync(ContainerImageTagsRequest);
+
+                    ErrorCheck(ContainerImageTagsResponse.Error);
+
+                    IEnumerable<string> existingImageTags = ContainerImageTagsResponse.Result.Tags.Where(x => x == settings.ContainerStartParameters.ImageDetails.ImageTag);
+
+                    if (existingImageTags.Any())
+                    {
+                        Console.WriteLine("Your container image is already uploaded. Continuing deployment...");
```

Comment:
FWIW, some folks like to overwrite the tags especially during development. This is a (very) bad practice IMHO, good thing that we are now allowing it.